### PR TITLE
RFC: Make trigonometric functions rigorous by calculating correctly the quadrants

### PR DESCRIPTION
**Pull request by dpsanders**

### CODE REVIEW

**Comment by dpsanders** [comment]
I think this is ready, @lbenet. Feel free to add more tests! Would you like me to rebase?

File: src/ValidatedNumerics.jl [hunk 1]
```diff
@@ -6,7 +6,7 @@ using Compat
 
 @compat round
 @compat parsefloat
-
+@compat ifloor
```
Comment (lbenet) [inline src/ValidatedNumerics.jl]: `ifloor` is deprecated in favor of `floor(Integer,x)`

File: src/intervals/interval_trig.jl [hunk 2]
```diff
@@ -1,113 +1,141 @@
 
 #----- From here on, NEEDS TESTING ------
 
+
+const float_pi_interval = @floatinterval(pi)
+const big_pi_interval = @interval(pi)
```
Comment (lbenet) [inline src/intervals/interval_trig.jl]: I guess there is some odd dependence `big_pi_interval` with regards to the default precision set (53). Probably all these constants should go into a type structure, which redefines them if the precision is changed.

File: src/root_finding/automatic_differentiation.jl [hunk 3]
```diff
@@ -40,6 +40,7 @@ end
 
 sin(x::Jet) = Jet(sin(x.val), x.der*cos(x.val))
 cos(x::Jet) = Jet(cos(x.val), -x.der*sin(x.val))
+tan(x::Jet) = Jet(tan(x.val), x.der / cos(x.val)^2 )
```
Comment (lbenet) [inline src/root_finding/automatic_differentiation.jl]: There is a `sec(x)` function, so this could be rewritten in a julian way as `Jet(tan(x.val), x.der * sec(x) ^2)`

Comment (dpsanders) [inline src/root_finding/automatic_differentiation.jl]: Since `sec(x)` is implemented directly as `one(T)/cos(x)`, I don't think that anything is gained by this (and in fact it could even be slightly less efficient).

File: src/intervals/interval_trig.jl (hunk 2 above)
Comment (dpsanders) [inline src/intervals/interval_trig.jl]: Yes, I agree. I propose to do this in a separate PR.
